src/dns.rs

```diff
@@ -125,6 +125,102 @@ cfg_if::cfg_if! {
     }
 }
 
+cfg_if::cfg_if! {
+    if #[cfg(target_os="macos")] {
+        use std::time::Duration;
+        use tokio::time::timeout;
+        use tokio::runtime;
+        use futures::stream::{StreamExt};
+        use async_dnssd::{query_record,QueryRecordResult,QueriedRecordFlags,Type};
+
+        #[derive(Clone)]
+        pub struct DnsSrvRecord {
+            priority: u16,
+            weight: u16,
+            port: u16,
+            target: String
+        }
+
+        impl From<&QueryRecordResult> for DnsSrvRecord {
+            fn from(record: &QueryRecordResult) -> Self {    
+                let rdata = record.rdata.as_slice();
+                let priority = u16::from_be_bytes(rdata[0..2].try_into().unwrap());
+                let weight = u16::from_be_bytes(rdata[2..4].try_into().unwrap());
+                let port = u16::from_be_bytes(rdata[4..6].try_into().unwrap());
+                let target_data = &rdata[6..rdata.len()];
+                DnsSrvRecord {
+                    priority: priority,
+                    weight: weight,
+                    port: port,
+                    target: dns_decode_target_data_to_string(target_data)
+                }
+            }
+        }
+
+        pub fn dns_decode_target_data_to_string(v: &[u8]) -> String {
+            let mut names = Vec::new();
+        
+            let mut i = 0;
+            while i < v.len() {
+                let size = v[i] as usize;
+                if size == 0 || i + 1 + size > v.len() {
+                    break;
+                }
+                names.push(String::from_utf8_lossy(&v[i+1..i+1+size]));
+                i = i + 1 + size;
+            }
+        
+            names.join(".")
+        }
+
+        pub fn dns_query_srv_records(name: &str) -> Vec<DnsSrvRecord> {
+            let query_timeout = 1000;
+            let mut dns_records: Vec<DnsSrvRecord> = Vec::new();
+        
+            let rt = runtime::Builder::new_current_thread().enable_all().build().unwrap();
```

Comment:
This never panics by auditing tokio code, maybe you can add a comment explaining that